db/c.cc

```diff
@@ -6037,7 +6038,7 @@ rocksdb_export_import_files_metadata_create() {
 
 char* rocksdb_export_import_files_metadata_get_db_comparator_name(
     rocksdb_export_import_files_metadata_t* metadata) {
-  return CopyString(metadata->rep->db_comparator_name);
+  return strdup(metadata->rep->db_comparator_name.c_str());
```

Comment:
This is the key change in this entire PR - I misunderstood how CopyString was used elsewhere. We should be returning a null-terminated string as it is more conventional and simpler for callers to deal with.